connectors/concur/src/main/java/com/vmware/connectors/concur/ConcurController.java

```diff
@@ -145,74 +151,81 @@ private String getConcurRequestTemplate(final String reason,
                                                       final String routingPrefix) {
         logger.debug("Requesting expense request info from concur base URL: {} for ticket request id: {}", baseUrl, id);
 
-        final ListenableFuture<ResponseEntity<JsonDocument>> result = this.rest.exchange("{baseUrl}/expense/reports/{id}",
+        final Single<ResponseEntity<JsonDocument>> result = getReportDetails(headers, id, baseUrl);
+        return result
+                .toObservable()
+                .onErrorResumeNext(ObservableUtil::skip404)
+                .map(entity -> convertResponseIntoCard(entity, baseUrl, id, routingPrefix));
+    }
+
+    private Single<ResponseEntity<JsonDocument>> getReportDetails(HttpHeaders headers, String id, String baseUrl) {
+        final ListenableFuture<ResponseEntity<JsonDocument>> result = this.rest.exchange("{baseUrl}/api/expense/expensereport/v2.0/report/{id}",
                 HttpMethod.GET,
                 new HttpEntity<String>(headers),
                 JsonDocument.class,
                 baseUrl,
                 id);
+        return Async.toSingle(result);
+    }
 
-        return Async.toSingle(result)
-                .toObservable()
-                .onErrorResumeNext(ObservableUtil::skip404)
-                .map(entity -> convertResponseIntoCard(entity, baseUrl, id, routingPrefix));
+    private Single<String> getWorkFlowActionUrl(final HttpHeaders headers,
+                                                final String id,
+                                                final String baseUrl) {
+        final Single<ResponseEntity<JsonDocument>> result = getReportDetails(headers, id, baseUrl);
```

Comment:
This is more concise:
```
      return getReportDetails(headers, id, baseUrl)
                .map(ResponseEntity::getBody)
                .map(jsonDocument -> jsonDocument.read("$.WorkflowActionURL"));
```

Reply:
I have changed the code to above format.